autoscaling/autoscaling.go

```diff
@@ -84,8 +84,10 @@ func (as *AutoScaling) query(params map[string]string, resp interface{}) error {
 		hreq.Header.Set("X-Amz-Security-Token", token)
 	}
 
-	signer := aws.NewV4Signer(as.Auth, "autoscaling", as.Region)
-	signer.Sign(hreq)
+	err = aws.SignV4(hreq, as.Auth, "autoscaling", as.Region.Name)
+	if err != nil {
```

Comment:
Not going to use `if err =... ; err != nil` - what's the benefit of doing it this way if you're returning immediately?